subsys/usb/usb_device.c

```diff
@@ -1389,12 +1437,22 @@ static int custom_handler(struct usb_setup_packet *pSetup,
 			continue;
 		}
 
-		/* An exception for AUDIO_CLASS is temporary and shall not be
-		 * considered as valid solution for other classes.
+		if (iface->custom_handler == NULL) {
+			continue;
+		}
+
+		/*
+		 * Audio has several interfaces.
+		 * if_descr points to the first interface, but the request
+		 * may be for subsequent ones, so forward each request to audio.
+		 * Audio custom handler has kind of observer role and we can
+		 * ignore the return value.
 		 */
-		if (iface->custom_handler &&
-		    (if_descr->bInterfaceNumber == (pSetup->wIndex & 0xFF) ||
-		     if_descr->bInterfaceClass == AUDIO_CLASS)) {
+		if (if_descr->bInterfaceClass == AUDIO_CLASS) {
+			(void)iface->custom_handler(pSetup, len, data);
+		}
+
+		if (if_descr->bInterfaceNumber == (pSetup->wIndex & 0xFF)) {
 			return iface->custom_handler(pSetup, len, data);
```

Comment:
Is this intentional that now the handler will be called twice for this audio interface?

Reply:
I think it can be moved to else branch now, I will change it.